src/NServiceBus.Core/Logging/DefaultFactory.cs

```diff
@@ -43,33 +45,48 @@ public void Level(LogLevel level)
         public void Directory(string directory)
         {
             Guard.AgainstNullAndEmpty(nameof(directory), directory);
+
             if (!IODirectory.Exists(directory))
             {
                 var message = $"Could not find logging directory: '{directory}'";
                 throw new DirectoryNotFoundException(message);
             }
+
             this.directory = new Lazy<string>(() => directory);
         }
 
-#if NET452
         internal static string FindDefaultLoggingDirectory()
         {
-            if (System.Web.HttpRuntime.AppDomainAppId == null)
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                if (assembly.GetName().Name == "System.Web")
+                {
+                    systemWebAssembly = assembly;
```

Comment:
since you're setting a static here, should we check the statics value before looping through the loaded assemblies first?

Another approach would be to not use a static and pass the assembly to DerivceAppDataPath and so on, but I don't see any issue if multiple endpoint instances would share the system.web assembly reference.